appmap/_implementation/event.py

```diff
@@ -183,9 +183,21 @@ def set_params(params, instance, args, kwargs):
         # dict, respectively. It operates on them as collections, so
         # it doesn't unpack them.
         ret = []
-        if instance is not None:
-            ret.append(params[0].to_dict(instance))
-            params = params[1:]
+
+        # HACK: this is to detect cases when this is a method, yet the self
+        # parameter is None. Unfortunately wrapt tries to be too smart
+        # by separating the instance argument, and as a consequence
+        # we can't differentiate here between a function and a method
+        # bound to None. Thus we rely on 'self' being the conventional
+        # name for the self argument. This is usually correct but
+        # theoretically could be wrong with code that's off-style.
+        takes_self = params and params[0].name == 'self'
```

Comment:
Theoretically, but I haven't seen that in the wild and I can't imagine how this could be useful; since this is a bit of a hack I decided to limit it to what was actually needed.